jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/ResourceHttpContentFactory.java

```
@@ -29,13 +29,13 @@
  */
 public class ResourceHttpContentFactory implements HttpContent.Factory
 {
-    private final ResourceFactory _factory;
+    private final Resource _baseResource;
     private final MimeTypes _mimeTypes;
 
-    public ResourceHttpContentFactory(ResourceFactory factory, MimeTypes mimeTypes)
+    public ResourceHttpContentFactory(Resource baseResource, MimeTypes mimeTypes)
     {
         Objects.requireNonNull(mimeTypes, "MimeTypes cannot be null");
-        _factory = factory;
+        _baseResource = Objects.requireNonNullElse(baseResource, ResourceFactory.root().newResource("."));
```

I like it that it is now explicit that the default base is the CWD, but isn't that a dangerously unsafe default? It feels like we should throw instead of having a default here.

---

It does feel a little dangerous... but it has always been the way that if no resourceBase is configured, then the CWD is used.      This cannot be the case in a jetty-home setup, but can be used in embedded jetty, and I have seen plenty of cases where people have relied on this behaviour.  

I think we could change it, but that feels like a 12.1 thing.